metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py

```diff
@@ -118,10 +139,12 @@ class AwsConnectionConfig(ConfigModel):
         description="Advanced AWS configuration options. These are passed directly to [botocore.config.Config](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html).",
     )
 
-    def allowed_cred_refresh(self) -> bool:
-        if self._normalized_aws_roles():
-            return True
-        return False
+    @staticmethod
+    def get_caller_identity(session: Session) -> Optional[str]:
```

Comment:
Can this change over time, or can we cache this?

Reply:
The logic of the ingestor itself does not need it, we extract the caller identity from the session to make sure we acquired correct role (in debug mode only)